### PR TITLE
Empty request list for non approvers

### PR DESCRIPTION
**Pull request by lgalis**
Fixes https://projects.engineering.redhat.com/browse/SSP-1563

Displays an empty list when the admin does not have an approval role.
Displays the not authorized page for a use with neither admin nor approver roles.
Clears list and metadata when switching between the request queue and all requests tab.

Admin user, no approval role:
Screenshot from 2020-06-04 16-33-20

Admin with approver role and approval processes assigned:
Screenshot from 2020-06-04 00-38-57

User with neither admin nor approver roles:
Screenshot from 2020-06-04 16-37-52

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@lgalis I think I don't fully understand this scenario:

Screenshot from 2020-06-04 03-53-57

I would expect this message when the API returns 0 results and no filters were applied. Instead, it shows this page when the admin does not have approval roles. I would expect some messages about missing roles or something similar. This is a bit confusing to me.

File: src/smart-components/request/request-detail/request.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ export const Request = ({ item, isExpanded, toggleExpand }) => {
   };
 
   const checkCapability = (item, capability) => {
-    if (isApprovalAdmin(userPersona)) {
+    if (isApprovalAdmin(userRoles)) {
```
Comment (Hyperkid123) [inline src/smart-components/request/request-detail/request.js:42]: ^^

Comment (Hyperkid123) [reply src/smart-components/request/request-detail/request.js:42]: These functions seem to me like good candidates for new hook aliases like `useIsApprovalAdmin` and others.

**Comment by lgalis** [comment]
@Hyperkid123 - the No requests you are asking about - is for the case when the admin does not have an approver role. 
The Request queue  is the list of requests a user has to approve based on belonging to groups in an approval process. If an admin does not have an approver role, this list will be empty.
The user will be assigned an approver role if a group it belongs to  is assigned to an approval process.

File: src/routing/requests-route.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, { useContext } from 'react';
+import { Route, Redirect, useLocation } from 'react-router-dom';
+import UserContext from '../user-context';
+import { isApprovalAdmin, isApprovalApprover } from '../helpers/shared/helpers';
+
+const RequestsRoute = (props) => {
```
[the remark this answers is not in the document]
Comment (lgalis) [reply src/routing/requests-route.js:6]: used in line 11

**Comment by sbuenafe-rh** [comment]
@lgalis Here's the PF guidance for the empty state: https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/empty-state 
It has which icons to use. 

For the 403 empty state message, 
We can keep the Approval header. 
The headline "Forbidden" should change to "You do not have access to Approval" 
CTA button: If they came from another page, the button should say “Return to previous page”. 
If they came to the page by directly inputting the URL, the button should say “Go to landing page”

**Comment by lgalis** [comment]
@sbuenafe-rh - updated per the docs. Screenshots updated in the PR's description.

**Pull request merged**